### PR TITLE
Rewards changes

### PR DESCRIPTION
**Pull request by yanivyehu**

### CODE REVIEW

File: contracts/RewardsSender.sol [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "./Bridge.sol";
```
Comment (israeldeweb) [inline contracts/RewardsSender.sol:5]: https://docs.soliditylang.org/en/v0.8.10/types.html#members-of-addresses

Take a look at the `call` function.

**Comment by israeldeweb** [comment]
Unify config number format. Remove trailing newline. Add semicolon at the end.

File: contracts/RewardsSender.sol [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+contract RewardsSender {
+    IBridge bridge;
+    IERC20 bbsToken;
+
+    constructor(IERC20 _bbsToken, IBridge _bridge) {
+        bridge = _bridge;
+        bbsToken = _bbsToken;
+        bbsToken.approve(address(bridge), 2**256 - 1);
```
Comment (israeldeweb) [inline contracts/RewardsSender.sol:17]: Note that this approval can still, theoretically, run out. I will just take about 10**68 transfers of our entire token supply :)

File: eosio/deployment/BancorX.abi [hunk 3]
```diff
@@ -88,6 +88,14 @@
                 {
                     "name": "max_destroy_limit",
                     "type": "uint64"
+                },
+                {
+                    "name": "rewards_receiver",
+                    "type": "name"
```
Comment (israeldeweb) [inline eosio/deployment/BancorX.abi:94]: The type of this argument is "name"? Is this a valid type, or a typo? And if it's a typo, does it still work? It appears in several places.

File: eosio/deployment/BancorX/BancorX.cpp [hunk 4]
```diff
@@ -0,0 +1,289 @@
+
```
Comment (israeldeweb) [inline eosio/deployment/BancorX/BancorX.cpp:1]: This and the hpp should be diffed against the version that Bancor is (still) running.

File: eosio/deployment/change_ownership.sh [hunk 5]
```diff
@@ -6,7 +6,11 @@ pushd "$(dirname "${BASH_SOURCE[0]}")"
 # Change the private key that controls a role in an account by giving a new public key.
```
Comment (israeldeweb) [inline eosio/deployment/change_ownership.sh:6]: Comment is no longer accurate. It's not "a role", it's the owner and active roles of a very specific eos account.

File: eosio/state_example.env [hunk 6]
```diff
@@ -0,0 +1,9 @@
+bbs_account='1vs5bschueot'
```
Comment (israeldeweb) [inline eosio/state_example.env:1]: Doesn't this info come from the js config now?

File: eosio/utils/create_accounts.sh [hunk 7]
```diff
@@ -66,7 +66,7 @@ if [ ! "$bridge_account" ]; then
         bridge_account="$(name_maker)"
         store_env bridge_account $bridge_account
 
-        bbs_active_public_key="$(cat state.env | grep "${bbs_account}_active_public_key" | sed 's/.*=//' | sed 's/^.//;s/.$//')"
+        bbs_active_public_key="$(cat ../state.env | grep "${bbs_account}_active_public_key" | sed 's/.*=//' | sed 's/^.//;s/.$//')"
```
Comment (israeldeweb) [inline eosio/utils/create_accounts.sh:69]: I do not like this. Can't `store_env` take care of this and set the variable while writing it to the file?

File: scripts/deployment/config.js [hunk 8]
```diff
@@ -10,15 +17,14 @@ module.exports = {
...
+        sendRewardsMaxLockLimit: '1000000000000000000000000'
     },
-    rewards: {
-        quartes: [0, 1, 2, 3, 4],
-        amount: '10000'
+    dailyRewards: {
+        //To avoid copying of rewardsSender address to config, we get it in the script directly
+        amount: 1000
     }
 }
```
Comment (israeldeweb) [inline scripts/deployment/config.js:29]: Missing semicolon and trailing newline.

File: scripts/deployment/config_example.js [hunk 9]
```diff
@@ -2,25 +2,32 @@
...
-        maxLockLimit : '40000000000000000000000',
-        maxReleaseLimit: '80000000000000000000000',
-        minLimit: '1000000000000000000',
-        limitIncPerBlock: '500000000000000000000',
-        minRequiredReports: 1,
-        commissionAmount: '0',
-        reporters: {
-            addresses: ['0x70997970c51812dc3a010c7d01b50e0d17dc79c8'],
-            active: [true]
+    "rewards": [ { "q" : 0, "amount": "100000"}, { "q" : 1, "amount": "100000"}],
```
Comment (israeldeweb) [inline scripts/deployment/config_example.js:12]: So numbers are still numbers, but amounts are strings, right? Perhaps you should add a comment at the top of the config.

File: scripts/deployment/daily_rewards_declare_rewards.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    const DailyRewards = await ethers.getContractFactory('DailyRewards');
+    const dailyRewards = DailyRewards.attach(DAILY_REWARDS_ADDRESS);
+
+    const tx = await dailyRewards.declareRewards([REWARDS_SENDER_ADDRESS], [ethers.utils.parseEther(config.dailyRewards.amount)]);
+    common.etherscanLogTx(tx.hash, tx.chainId);
+    common.writeArtifact(ARTIFACT_FILE, JSON.stringify(tx));
+    log(`---Declare rewards Done---`);
+}
+
+main().then(() => process.exit(0)).catch(error => {
```
Comment (israeldeweb) [inline scripts/deployment/daily_rewards_declare_rewards.js:27]: This snippet repeats at the end of so many scripts, I'm considering adding a `run` function in `common.js` so we can replace it with something like `await run(main);`.

File: scripts/deployment/deploy_bridge.js [hunk 11]
```diff
@@ -16,18 +16,27 @@ async function main() {
...
+        hardhat.ethers.utils.parseEther(config.bridge.minLimit),
+        hardhat.ethers.utils.parseEther(config.bridge.limitIncPerBlock),
         config.bridge.minRequiredReports,
-        config.bridge.commissionAmount,
+        hardhat.ethers.utils.parseEther(config.bridge.commissionAmount),
+        sendRewardsData,
         BBS_TOKEN_ADDRESS);
     common.etherscanLogContract(bridge.address, bridge.deployTransaction.chainId);
     fs.writeFileSync(common.bridgePath, bridge.address);
+    fs.writeFileSync(common.bridgeSendRewardsArgPath, sendRewardsData);
```
Comment (israeldeweb) [inline scripts/deployment/deploy_bridge.js:39]: This whole shifting between js config file and env files and stuff starts to smell a bit. I think you hinted at it when you asked for the review. Let's go over the mechanism on the phone.

File: scripts/deployment/deploy_daily_rewards.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
+const fs = require('fs');
```
Comment (israeldeweb) [inline scripts/deployment/deploy_daily_rewards.js:1]: A lot of code duplication. Maybe a semi-generic contract deployer in `common.js`?

File: scripts/tests/deployment_test.js [hunk 13]
```diff
@@ -0,0 +1,225 @@
+/**
```
Comment (israeldeweb) [inline scripts/tests/deployment_test.js:1]: I assume you just moved the file. Next time (or, if you feel like it, you can fix it with a quick rebase) use `git mv` instead, I think github will show us the diff automatically.

File: scripts/tests/staking_scenario.js [hunk 14]
```diff
@@ -1,10 +1,10 @@
-const common = require('./common/common');
-const {range} = require('../test/utils');
+const common = require('../common/common');
```
Comment (israeldeweb) [inline scripts/tests/staking_scenario.js:1]: Did you do a `git mv` here? That's what I wanted to see in `deployment_test.js`.

File: test/dailyRewards-test.js [hunk 15]
```diff
@@ -1,8 +1,10 @@
 const {expect} = require('chai');
 const {expectRevert} = require('./utils');
+const { network } = require('hardhat');
```
Comment (israeldeweb) [inline test/dailyRewards-test.js:3]: Inconsistent spacing inside `{}`.

File: contracts/RewardsSender.sol (hunk 2 above)
Comment (yanivyehu) [reply contracts/RewardsSender.sol:17]: we will be rich much sooner :)

File: eosio/deployment/BancorX.abi (hunk 3 above)
Comment (yanivyehu) [reply eosio/deployment/BancorX.abi:94]: yes this is a type.

Comment (yanivyehu) [reply eosio/deployment/BancorX.abi:94]: https://developers.eos.io/manuals/eosio.cdt/v1.8/structeosio_1_1name

Comment (yanivyehu) [reply eosio/deployment/BancorX.abi:94]: it will no pass compilation if this is a typo...

File: eosio/deployment/BancorX/BancorX.cpp (hunk 4 above)
Comment (yanivyehu) [reply eosio/deployment/BancorX/BancorX.cpp:1]: this is the commit you are looking for:
https://github.com/deweb-io/token/commit/0444cb5a8aa7407df32af64359551bae468aefd7

File: scripts/deployment/deploy_bridge.js (hunk 11 above)
Comment (yanivyehu) [reply scripts/deployment/deploy_bridge.js:39]: let's.

File: scripts/tests/deployment_test.js (hunk 13 above)
Comment (yanivyehu) [reply scripts/tests/deployment_test.js:1]: yes. I just moved it. next time will do it with git mv.

File: scripts/tests/staking_scenario.js (hunk 14 above)
Comment (yanivyehu) [reply scripts/tests/staking_scenario.js:1]: no git mv.

File: test/dailyRewards-test.js (hunk 15 above)
Comment (yanivyehu) [reply test/dailyRewards-test.js:3]: OMG :)

File: scripts/deployment/deploy_daily_rewards.js (hunk 12 above)
Comment (yanivyehu) [reply scripts/deployment/deploy_daily_rewards.js:1]: not sure about that.
different validations, different arguments to constructor, and staking using 'deployProxy'.

File: eosio/state_example.env (hunk 6 above)
Comment (yanivyehu) [reply eosio/state_example.env:1]: accounts of eos should be in state.env

File: eosio/utils/create_accounts.sh (hunk 7 above)
Comment (yanivyehu) [reply eosio/utils/create_accounts.sh:69]: only the public key of bbs is read from state.env, so I don’t see why to change store_env.

File: scripts/deployment/daily_rewards_declare_rewards.js (hunk 10 above)
Comment (yanivyehu) [reply scripts/deployment/daily_rewards_declare_rewards.js:27]: nice to have

**Pull request merged**